tests/common.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from abc import ABC
+
+import yaml
+
+from mewbot.loader import load_component
+from mewbot.core import ComponentKind
+from mewbot.component import ConfigBlock
+from mewbot.io.http import HTTPServlet
+
+
+class BaseTestClassWithConfig(ABC):
+    # pylint: disable=R0903
+    # Too few public methods, because this is always used elsewhere
+    fname: str
+    config: ConfigBlock
+    doctype: ComponentKind
+    config_toload = True
+
+    def cache_config(self) -> None:
+        if self.config_toload:
+            with open(self.fname, "r", encoding="utf-8") as config:
+                for document in yaml.load_all(config, Loader=yaml.CSafeLoader):
+                    if document["kind"] == self.doctype:
+                        self.config = document
+                        self.config_toload = False
```

Comment:
I'd invert the condition here to reduce indentation, and also exit out once we match an item
 ```suggestion
        if not self.config_toload:
            return
        with open(self.fname, "r", encoding="utf-8") as config:
            for document in yaml.load_all(config, Loader=yaml.CSafeLoader):
                if document["kind"] == self.doctype:
                    self.config = document
                    self.config_toload = False
                    return
```